MQTTClient/MQTTClient/MQTTWebsocketTransport/MQTTWebsocketTransport.m

```diff
@@ -34,8 +36,15 @@ - (instancetype)init {
 - (void)open {
     DDLogVerbose(@"[MQTTWebsocketTransport] open");
     self.state = MQTTTransportOpening;
+    NSMutableURLRequest *urlRequest = nil;
```

Comment:
Instead of this, can we check:

if `url == nil` inside `endpointURL` and return it, otherwise do the same?